client/src/app/shared/components/slide-container/slide-container.component.ts

```diff
@@ -209,6 +209,7 @@ export class SlideContainerComponent extends BaseComponent {
     private setDataForComponent(): void {
         if (this.slideRef && this.slideRef.instance) {
             this.slideRef.instance.data = this.slideData;
+            this.slideRef.changeDetectorRef.detectChanges();
```

Comment:
don't expose `changeDetectorRef` publicly. It's cleaner to use a function in `slideRef` than to change methods here.

However: This should not be necessary, if all changes to inner Slides are observed already, no?

Reply:
`slideRef` has the type `ComponentRef` which explicitly exposes the CDRef: https://angular.io/api/core/ComponentRef

I do not know the internals too much, but I guess that in this case the normal change detection is circumvented. As far as I looked into angulars source code, `@Input()` is just a marker and does not trigger the CD when the value is changed.